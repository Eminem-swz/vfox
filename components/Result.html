<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Result 结果 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.06f325cd.css">
    <link rel="modulepreload" href="/vfox/assets/Home.cf6c77ab.js">
    <link rel="modulepreload" href="/vfox/assets/app.8ec95c28.js">
    <link rel="modulepreload" href="/vfox/assets/components_Result.md.b5e0629d.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.8ec95c28.js">
    <meta name="twitter:title" content="Result 结果 | Vfox for Vue3">
    <meta property="og:title" content="Result 结果 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/LocalStorage" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/LocalStorage" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Result">Result</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#附加内容（-default）">附加内容（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><!-- TODO: make this button accessible --><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/Result" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="result-结果" tabindex="-1">Result 结果 <a class="header-anchor" href="#result-结果" aria-hidden="true">#</a></h1><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>type</td><td>string</td><td>&#39;info&#39;</td><td>否</td><td>类型，对应不同的提示图标，可选 &#39;info&#39;, &#39;warning&#39;, &#39;success&#39;, &#39;fail&#39;</td></tr><tr><td>title</td><td>string</td><td>&#39;&#39;</td><td>否</td><td>标题</td></tr><tr><td>description</td><td>string</td><td>&#39;&#39;</td><td>否</td><td>小标题，副标题，描述</td></tr><tr><td>show-back</td><td>boolean</td><td>true</td><td>否</td><td>是否显示返回按钮</td></tr><tr><td>confirm-text</td><td>string</td><td>&#39;确定&#39;</td><td>否</td><td>确认按钮的文字</td></tr><tr><td>back-text</td><td>string</td><td>&#39;返回&#39;</td><td>否</td><td>返回按钮的文字</td></tr></tbody></table><h2 id="slots" tabindex="-1">Slots <a class="header-anchor" href="#slots" aria-hidden="true">#</a></h2><h3 id="附加内容（-default）" tabindex="-1">附加内容（#default） <a class="header-anchor" href="#附加内容（-default）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-result
  type=&quot;fail&quot;
  title=&quot;支付失败&quot;
  description=&quot;&quot;
&gt;
  &lt;div class=&quot;result-extra&quot;&gt;
    在这里可以附加组件元素或者相应的提示文案，样式自定。
  &lt;/div&gt;
&lt;/fx-result&gt;
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/Stopwatch" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>Stopwatch</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/LoadMore" data-v-38ede35f><span class="text" data-v-38ede35f>LoadMore</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_drawer.md\":\"1d00d4eb\",\"apis_intersectionobserver.md\":\"ada6c900\",\"apis_localstorage.md\":\"27c2f94d\",\"apis_modal.md\":\"3ed48854\",\"apis_notify.md\":\"7e83d7e3\",\"apis_popover.md\":\"3b2557c1\",\"apis_preview.md\":\"917d4a62\",\"apis_scroll.md\":\"ec702526\",\"apis_selectorquery.md\":\"df90dfa7\",\"apis_toast.md\":\"14e6b224\",\"apis_index.md\":\"79305c68\",\"components_actionsheet.md\":\"5e5c30bc\",\"components_activityindicator.md\":\"cc9066e4\",\"components_avatar.md\":\"eaea51fa\",\"components_backtop.md\":\"9a78e54e\",\"components_badge.md\":\"1334d501\",\"components_button.md\":\"80f375a3\",\"components_calendar.md\":\"9c139854\",\"components_cascader.md\":\"66ec37f9\",\"components_cell.md\":\"28f35fe1\",\"components_checkbox.md\":\"f9b1615a\",\"components_circleprogress.md\":\"df034b7b\",\"components_collapse.md\":\"68497691\",\"components_copy.md\":\"647c421c\",\"components_countdown.md\":\"080ddff6\",\"components_countup.md\":\"e5699d8c\",\"components_datepicker.md\":\"7b7ac87c\",\"components_dialog.md\":\"a391eaf6\",\"components_divider.md\":\"902adb25\",\"components_drawer.md\":\"d0bc7cbc\",\"components_dropdown.md\":\"155d1eb2\",\"components_empty.md\":\"7d257847\",\"components_fixed.md\":\"4f518e33\",\"components_flatlist.md\":\"8ecf0c15\",\"components_form.md\":\"e2d5c597\",\"components_group.md\":\"1ed9eae8\",\"components_icon.md\":\"dd265250\",\"components_image.md\":\"47ab8041\",\"components_imagepreview.md\":\"ce73791e\",\"components_imageuploader.md\":\"0ecf839a\",\"components_indexview.md\":\"c0f07f00\",\"components_input.md\":\"ba30f694\",\"components_layout.md\":\"829392fe\",\"components_loadmore.md\":\"186e8b04\",\"components_modal.md\":\"fc4fe438\",\"components_navbar.md\":\"c93caeff\",\"components_noticebar.md\":\"709b90bc\",\"components_notify.md\":\"cdf9bd3c\",\"components_numberkeyboard.md\":\"9b794d0c\",\"components_order.md\":\"998ababc\",\"components_pagination.md\":\"0aa24940\",\"components_picker.md\":\"d917dac6\",\"components_popdialog.md\":\"3a517ad2\",\"components_popmenu.md\":\"feaa8680\",\"components_popover.md\":\"50139339\",\"components_price.md\":\"555a9b2e\",\"components_progress.md\":\"5efeb800\",\"components_radio.md\":\"1b53d2c2\",\"components_range.md\":\"d5a60765\",\"components_rate.md\":\"62ccde4d\",\"components_result.md\":\"b5e0629d\",\"components_scrolltab.md\":\"131b9ef5\",\"components_scrollview.md\":\"e1165ceb\",\"components_searchbar.md\":\"9ddb5927\",\"components_sidetab.md\":\"e1cbd95a\",\"components_skeleton.md\":\"69b49669\",\"components_slider.md\":\"7794dfc9\",\"components_stepper.md\":\"6ac8a7dd\",\"components_steps.md\":\"47a3f6fc\",\"components_sticky.md\":\"1cbfa2a3\",\"components_stopwatch.md\":\"45da4906\",\"components_swipecell.md\":\"851224e8\",\"components_swiper.md\":\"ec45a831\",\"components_switch.md\":\"8f83a880\",\"components_tab.md\":\"21bfc5a0\",\"components_tabbar.md\":\"b1b659f9\",\"components_tabview.md\":\"42249e03\",\"components_tag.md\":\"96e8e2c3\",\"components_timeago.md\":\"fc75a443\",\"components_timeline.md\":\"10cdacc1\",\"components_toast.md\":\"c5445cb5\",\"components_index.md\":\"bacb91bc\",\"design_index.md\":\"f0d612d3\",\"index.md\":\"a7dd64e7\"}")</script>
    <script type="module" async src="/vfox/assets/app.8ec95c28.js"></script>
    
  </body>
</html>