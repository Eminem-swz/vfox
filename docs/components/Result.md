# Result 结果

## Import

```
import { Result } from 'vfox'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

## Props

| 属性         | 类型    | 默认值 | 必填 | 说明                                                                |
| ------------ | ------- | ------ | ---- | ------------------------------------------------------------------- |
| type         | string  | 'info' | 否   | 类型，对应不同的提示图标，可选 'info', 'warning', 'success', 'fail' |
| title        | string  | ''     | 否   | 标题                                                                |
| description  | string  | ''     | 否   | 小标题，副标题，描述                                                |
| show-back    | boolean | true   | 否   | 是否显示返回按钮                                                    |
| confirm-text | string  | '确定' | 否   | 确认按钮的文字                                                      |
| back-text    | string  | '返回' | 否   | 返回按钮的文字                                                      |

## Slots

### 附加内容（#default）

```
<fx-result
  type="fail"
  title="支付失败"
  description=""
>
  <div class="result-extra">
    在这里可以附加组件元素或者相应的提示文案，样式自定。
  </div>
</fx-result>
```
