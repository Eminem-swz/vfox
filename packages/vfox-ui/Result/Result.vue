<template>
  <div class="fx-result" :class="['type--' + type2]">
    <div class="fx-result_header">
      <Icon class="fx-result_icon" :class="['type--' + type2]" :icon="icon" />
      <div class="fx-result_title" v-if="title">{{ title }}</div>
      <div class="fx-result_description" v-if="description">
        {{ description }}
      </div>
    </div>
    <slot></slot>
    <div class="fx-result_footer">
      <Button type="primary" @click="onConfirmClick">
        {{ confirmText || locale.resultConfirmText }}
      </Button>
      <Button v-if="showBack" type="default" @click="onCancelClick">
        {{ backText || locale.resultBackText }}
      </Button>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue'
import type { PropType } from 'vue'
import { Icon } from '../Icon'
import { Button } from '../Button'
import {
  createEnumsValidator,
  emitEventValidator,
  getEnumsValue
} from '../helpers/validator'
import { locale } from '../Locale'

type ResultType = 'info' | 'warning' | 'success' | 'fail'

const iconMap = new Map<ResultType, string>([
  ['info', 'InfoCircleFilled'],
  ['warning', 'WarningFilled'],
  ['success', 'CheckCircleFilled'],
  ['fail', 'CloseCircleFilled']
])

const RESULT_TYPES: ResultType[] = ['info', 'warning', 'success', 'fail']

export default defineComponent({
  name: 'fx-result',
  components: { Icon, Button },
  props: {
    // 类型
    type: {
      type: String as PropType<ResultType>,
      validator: createEnumsValidator(RESULT_TYPES),
      default: null
    },
    // 标题
    title: {
      type: String,
      default: ''
    },
    // 描述
    description: {
      type: String,
      default: ''
    },
    // 展示返回按钮
    showBack: {
      type: Boolean,
      default: true
    },
    // 返回文本文本
    backText: {
      type: String
    },
    // 确认文本
    confirmText: {
      type: String
    }
  },
  emits: {
    confirm: emitEventValidator,
    back: emitEventValidator
  },
  setup(props, { emit }) {
    const type2 = computed(() => getEnumsValue(RESULT_TYPES, props.type))

    const icon = computed(() => iconMap.get(type2.value) as string)

    function onConfirmClick(e: Event) {
      emit('confirm', e)
    }

    function onCancelClick(e: Event) {
      emit('back', e)
    }

    return {
      type2,
      icon,
      onConfirmClick,
      onCancelClick,
      locale
    }
  }
})
</script>
