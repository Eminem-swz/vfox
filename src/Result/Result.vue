<template>
  <div class="fx-result" :class="['type--' + type2]">
    <div class="fx-result_header">
      <icon
        class="fx-result_icon"
        :class="['type--' + type2]"
        :icon="icon"
      ></icon>
      <div class="fx-result_title" v-if="title">{{ title }}</div>
      <div class="fx-result_description" v-if="description">
        {{ description }}
      </div>
    </div>
    <slot></slot>
    <div class="fx-result_footer">
      <fx-button type="primary" @click="onConfirmClick">
        {{ confirmText }}
      </fx-button>
      <fx-button v-if="showBack" type="default" @click="onCancelClick">
        {{ backText }}
      </fx-button>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, PropType } from 'vue'
import Icon from '@/Icon'
import FxButton from '@/Button'
import { createEnumsValidator, getEnumsValue } from '@/helpers/validator'

type ResultType = 'info' | 'warning' | 'success' | 'fail'

const iconMap = new Map<ResultType, string>([
  ['info', 'InfoCircleFilled'],
  ['warning', 'WarningFilled'],
  ['success', 'CheckCircleFilled'],
  ['fail', 'CloseCircleFilled']
])

const RESULT_TYPES: ResultType[] = ['info', 'warning', 'success', 'fail']

export default defineComponent({
  name: 'fx-result',
  components: { Icon, FxButton },
  props: {
    // 类型
    type: {
      type: String as PropType<ResultType>,
      validator: createEnumsValidator(RESULT_TYPES),
      default: null
    },
    // 标题
    title: {
      type: String,
      default: ''
    },
    // 描述
    description: {
      type: String,
      default: ''
    },
    // 展示返回按钮
    showBack: {
      type: Boolean,
      default: true
    },
    // 返回文本文本
    backText: {
      type: String,
      default: '返回'
    },
    // 确认文本
    confirmText: {
      type: String,
      default: '确定'
    }
  },
  emits: ['confirm', 'back'],
  setup(props, { emit }) {
    const type2 = computed(() => getEnumsValue(RESULT_TYPES, props.type))

    const icon = computed(() => iconMap.get(type2.value))

    function onConfirmClick() {
      emit('confirm', {
        type: 'confirm'
      })
    }

    function onCancelClick() {
      emit('back', {
        type: 'back'
      })
    }

    return {
      type2,
      icon,
      onConfirmClick,
      onCancelClick
    }
  }
})
</script>
