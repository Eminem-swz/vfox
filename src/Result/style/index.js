import '../../Icon/style/index'
import '../../Button/style/index'
import '../index.scss'
