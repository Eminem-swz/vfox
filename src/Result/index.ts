import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import Result from './Result.vue'

const _Result: SFCWithInstall<typeof Result> = Object.assign(Result, {
  install: function(app: App) {
    app.component(Result.name, Result)
  }
})

export default _Result
