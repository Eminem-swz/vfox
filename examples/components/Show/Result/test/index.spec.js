/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import Result from '@/Result'

describe('Result', () => {
  test('should render default correctly', () => {
    const wrapper = shallowMount(Result, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
