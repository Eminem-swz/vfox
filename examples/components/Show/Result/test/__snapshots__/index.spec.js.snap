// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render default correctly 1`] = `
<div class="fx-result type--info">
  <div class="fx-result_header">
    <icon-stub icon="InfoCircleFilled" spin="false" size="0" width="0" height="0" class="fx-result_icon type--info"></icon-stub>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="fx-result_footer">
    <fx-button-stub type="primary" loading="false" disabled="false" ghost="false" transparent="false"></fx-button-stub>
    <fx-button-stub type="default" loading="false" disabled="false" ghost="false" transparent="false"></fx-button-stub>
  </div>
</div>
`;
